app/controllers/api/v2/config_templates_controller.rb

```diff
@@ -79,9 +79,10 @@ def destroy
         process_response @config_template.destroy
       end
 
-      api :GET, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
+      api :POST, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
 
       def build_pxe_default
+        ::ActiveSupport::Deprecation.warn('GET method for build pxe default is deprecated. Please use POST instead.') if request.method == "GET"
```

Comment:
I think it's not exactly what @domcleal  has meant, we should leave the GET variant, but before doing its work, it should log a deprecation warning.
something like this:
``` Ruby
api :GET, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
def build_pxe_default_with_warning
  Deprecation.warn('...')
  build_pxe_default
end

api :POST, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
def build_pxe_default
  do_work here
end
```